ros2cli/ros2cli/node/daemon.py

```diff
@@ -72,11 +72,14 @@ def spawn_daemon(args, wait_until_spawned=None):
             'Unable to get rmw_implementation_identifier, '
             'try specifying the implementation to use via the '
             "'RMW_IMPLEMENTATION' environment variable")
-    subprocess.Popen(cmd + [
+    cmd.extend([
         # the arguments are only passed for visibility in e.g. the process list
         '--rmw-implementation', rmw_implementation_identifier,
-        '--ros-domain-id', str(ros_domain_id)],
-        stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, **kwargs)
+        '--ros-domain-id', str(ros_domain_id)])
+    if not hasattr(args, 'debug') or not args.debug:
```

Comment:
@ivanpauno hmm all CLIs that use `NodeStrategy` and happen to have a `debug` option may unintentionally hook up the daemon to their standard streams. Consider adding another argument to the `spawn_daemon` function instead.

Reply:
See https://github.com/ros2/ros2cli/pull/284/commits/e33912323089920a28a992bc1410d56d1ee014c1.